packages/cra-template-mephisto-review/template/src/components/WordCloud.jsx

```diff
@@ -0,0 +1,214 @@
+      ? Math.ceil(
+          ((num - in_min) * (out_max - out_min)) / (in_max - in_min) + out_min
+        )
+      : ((num - in_min) * (out_max - out_min)) / (in_max - in_min) + out_min;
+  };
+
+  //reduces deviations between word counts given input of an object representing string words mapped to word count integers
+  // and maps word counts to em font sizes
+  const scaleEntries = (wordDict) => {
+    //get all words recorded
+    var words = Object.keys(wordDict);
+    //if empty, abort
+    if (!words || words.length == 0) return {};
+    //find min and max of all words
+    var values = Object.values(wordDict);
+    var maxVal = Math.max.apply(Math, values);
+    var minVal = Math.min.apply(Math, values);
+
+    if (maxVal - minVal > 10) {
+      //if difference between max and min word counts are over 10 remove all min counts
+      for (const word of words) {
+        if (wordDict[word] == minVal) delete wordDict[word];
+      }
+
+      //recalculate words, values, min and max
+      words = Object.keys(wordDict);
+      values = Object.values(wordDict);
+      maxVal = Math.max.apply(Math, values);
+      minVal = Math.min.apply(Math, values);
+    }
+
+    //calculate average word count
+    var sum = 0;
+    for (const word of words) {
+      sum += wordDict[word];
+    }
+    const avg = Math.ceil(sum / words.length);
+
+    //set maximum accepted word count to difference between average and smallest value in list
+    const diff = avg - minVal;
+    const maxRange = avg + diff;
+
+    //reduce all values over maximum accepted word count to the maximum accepted word count
+    for (const word of words) {
+      wordDict[word] = Math.min(wordDict[word], maxRange);
+    }
+
+    //maps word counts to em font sizes
+    for (const word of words) {
+      wordDict[word] = scaleNumber(
+        wordDict[word],
+        1,
+        maxRange,
+        minFontEmSize,
+        maxFontEmSize
+      );
+    }
+    return wordDict;
+  };
+
+  //get all word counts
+  var wordCounts = {};
+  switch (typeof data) {
+    case "object":
+      wordCounts = getAllObjectValWordCounts(data);
+      break;
+    case "string":
+      wordCounts = getAllStringWordCounts(data);
+      break;
+    default:
+      break;
+  }
+
+  //normalize word counts and map to em font sizes
+  wordCounts = scaleEntries(wordCounts);
+  const wordKeys = Object.keys(wordCounts);
+
+  //replace em font sizes in list with css styles for spans
+  for (const word of wordKeys) {
+    const size = wordCounts[word];
+    wordCounts[word] = {
+      fontSize: size + "em",
+      fontWeight: scaleNumber(
+        size,
+        minFontEmSize,
+        maxFontEmSize,
+        minFontWeight,
+        maxFontWeight,
+        true
+      ),
+    };
+  }
+
+  //map css styles to spans with corresponding word
+  var spans = [];
+  for (let word of wordKeys) {
+    const style = wordCounts[word];
+    spans.push(<span style={style}>{" " + word + " "}</span>);
+  }
```

Comment:
this whole loops above seems unnecessary and makes the code longer for no good reason. you could just inline this into the map below:

```jsx
return (
    <div className="cloud">
      <p>{wordKeys.map(word => <span style={wordCounts[word]}>{" " + word + " "}</span>)}</p>
    </div>
  );
```

Reply:
Oooh, thought crossed my mind, forgot to condense it, nice catch!